File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/ticket-sales/types/api/segment.ts
import { SegmentType } from '../enums/TicketSales';

export interface SegmentResponse {
  segmentId: number;
  name?: string;
  description?: string;
  capacity: number;
  segmentType: SegmentType;
  venueId: number;
  zoneIds?: number[];
}

File: MusicEventManagementSystem/MusicEventManagementSystem.Core/Models/Entities/TicketSales/Segment.cs
using MusicEventManagementSystem.Core.Enums.TicketSales;

namespace MusicEventManagementSystem.Core.Models.Entities.TicketSales
{
    public class Segment
    {
        public int SegmentId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public int Capacity { get; set; }
        public SegmentType SegmentType { get; set; }

        public int VenueId { get; set; }
        public Venue Venue { get; set; } = null!;

        public ICollection<Zone> Zones { get; set; } = new List<Zone>();
    }
}
